docs/sdk/examples/assets/pastefromoffice/pastefromoffice.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+ var sources = [
+    {
+        name: 'Google Docs',
+        link: 'pastefromgoogledocs.html',
+        linkName: 'Paste from Google Docs',
+        regex: /id=(\"|\')?docs\-internal\-guid\-/,
+        isAlreadyShow: false
+    },
+    {
+        name: 'Microsoft Word',
+        link: 'pastefromword.html',
+        linkName: 'Paste from Word',
+        regex: getProperWordRegex(),
+        isAlreadyShow: false
+    },
+    {
+        name: 'Microsoft Excel',
+        link: 'pastefromexcel.html',
+        linkName: 'Paste from Excel',
+        regex: /<meta\s*name=generator\s*content="Microsoft\s*Excel\s*\d+"?\/?/gmi,
+        isAlreadyShow: false
+    },
+    {
+        name: 'LibreOffice',
+        link: 'pastefromlibreoffice.html',
+        linkName: 'Paste from LibreOffice',
+        regex: /<meta\s+name=["']?generator["']?\s+content="LibreOffice\s*\d+"?\/?/gmi,
+        isAlreadyShow: false
+    }
+];
+
+CKEDITOR.on( 'instanceReady', function( evt ) {
+	var editor = evt.editor;
+
+	editor.on( 'paste', function( evt ) {
+		var pasteData = getClipboardData( evt.data, 'text/html' );
+		if( !pasteData ){
+			return;
+		}
+
+		for ( var i = 0; i < sources.length; i++ ) {
+			var source = sources[ i ];
+
+			if( source.regex.test( pasteData ) && !sources[ i ].isAlreadyShow ) {
+				var notice = document.querySelector( '.main__notification.notice' );
+
+				// Remove notification when user trying to paste content from other source than previous. e.g. Word and Excel.
+				if( notice ){
+					notice.remove();
+				}
+
+				createClipboardInputNotification( i );
+				sources[ i ].isAlreadyShow = true;
+				break;
+			}
+		}
+	}, null, null, 9 );
```

Comment:
I see that setting lower priority worked 🎉 Could you add a small comment on why it's needed? I.e. that `paste` event gets canceled by some plugin custom paste handlers like for `easyimage`?